editor/store/persist.js

```diff
@@ -34,17 +34,17 @@ export function withRehydratation( reducer, reducerKey ) {
  *
  * This should be executed after the reducer's registration.
  *
- * @param {Object} store      Store to enhance.
- * @param {string} reducerKey The reducer key to persist (example: reducerKey.subReducerKey).
- * @param {string} storageKey The storage key to use.
- * @param {Object} defaults   Default values of the reducer key.
+ * @param {Object}   store      Store to enhance.
+ * @param {Function} reducer    The reducer function. Used to get default values and to allow custom serialization by the reducers.
+ * @param {string}   reducerKey The reducer key to persist (example: reducerKey.subReducerKey).
+ * @param {string}   storageKey The storage key to use.
  */
-export function loadAndPersist( store, reducerKey, storageKey, defaults = {} ) {
+export function loadAndPersist( store, reducer, reducerKey, storageKey ) {
```

Comment:
This feels very much like it would fit the signature of a [store enhancer](https://redux.js.org/docs/Glossary.html#store-enhancer):

```js
export function loadAndPersist( reducerKey, storageKey ) {
	return ( createStore ) => ( reducer, ...args ) => {
		const store = createStore( reducer, ...args );

		// ...

		return store;
	};
}
```

_(Reference: [`applyMiddleware`](https://github.com/reactjs/redux/blob/master/src/applyMiddleware.js))_

But I'd guess this isn't well-supportable by our data module?

Reply:
Yes, I remember that some enhancers had to be removed because of not being well-supported.
@youknowriad, are you aware if the situation changed and now we can safely use a store enhancer?